nifi-nar-bundles/nifi-splunk-bundle/nifi-splunk-processors/src/main/java/org/apache/nifi/processors/splunk/PutSplunk.java

```diff
@@ -185,7 +185,7 @@ public void process(final InputStream in) throws IOException {
         // if TCP and we don't end in a new line then add one
         final String protocol = context.getProperty(PROTOCOL).getValue();
         if (protocol.equals(TCP_VALUE.getValue())) {
-            final byte[] buf = baos.getUnderlyingBuffer();
+            final byte[] buf = baos.toByteArray();
```

Comment:
actually i don't think it will break it.  however, we should take care to ensure we only create the buffer copy one time when we don't have to add the newline and twice when we do . Right now it would get done twice always as shown here

  https://github.com/apache/nifi/blob/70ede6ecd97d344d9883a859dd4974141cba4224/nifi-nar-bundles/nifi-splunk-bundle/nifi-splunk-processors/src/main/java/org/apache/nifi/processors/splunk/PutSplunk.java#L188

and here 
  https://github.com/apache/nifi/blob/70ede6ecd97d344d9883a859dd4974141cba4224/nifi-nar-bundles/nifi-splunk-bundle/nifi-splunk-processors/src/main/java/org/apache/nifi/processors/splunk/PutSplunk.java#L201


Thanks